### ISSUE TITLE
Document `nunjucks.parser`

### ISSUE THREAD

**Issue by ogonkov** [mozilla/nunjucks#1239]
It's a hidden treasure, and it would be nice to have some docs for `nunjucks.parser` and `nunjucks.nodes.*`.

You could guess the meaning of some `nunjucks.nodes`, but having docs would be much better.

**Comment by fdintino** [comment]
I agree this is a good idea, but some of the API will likely change for v4 of nunjucks (currently in-progress). I'll write some docs for it as part of the release.

**Comment by ogonkov** [comment]
@fdintino where i can track v4 progress? (develop)[https://github.com/mozilla/nunjucks/compare/develop] branch?

**Comment by fdintino** [comment]
@ogonkov that is where I'll be pushing out changes. I have a bunch of work locally that I need to clean up and push out. I may find time to do it this weekend.

**Comment by ogonkov** [comment]
Should be done after #1059
